src/Factory/RepositoryContainer.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Cycle\Factory;
+
+use Closure;
+use Cycle\ORM\ORMInterface;
+use Cycle\ORM\RepositoryInterface;
+use Cycle\ORM\SchemaInterface;
+use Psr\Container\ContainerInterface;
+use Yiisoft\Di\Container;
+use Yiisoft\Di\CompositeContainer;
+use Yiisoft\Di\Contracts\ServiceProviderInterface;
+
+use Yiisoft\Factory\Exception\NotFoundException;
+use function is_string;
+
+final class RepositoryContainer implements ContainerInterface
+{
+    private ContainerInterface $rootContainer;
+    private array $repositoryFactories = [];
+    private array $instances;
+    private bool $build = false;
+
+    public function __construct(ContainerInterface $rootContainer)
+    {
+        $this->rootContainer = $rootContainer;
+    }
+
+
+    public function get($id)
+    {
+        if (isset($this->instances[$id])) {
+            return $this->instances[$id];
+        }
+
+        if ($this->has($id)) {
+            return $this->instances[$id] = $this->repositoryFactories[$id]();
+        }
+
+        throw new NotFoundException("Repository $id doesn't exist.");
+    }
+
+    public function has($id): bool
+    {
+        if (!$this->build) {
```

Comment:
You should make container more lazy.
Before building do checking that $id is subclass of `RepositoryInterface`

Reply:
Fixed